### PR TITLE
use from_blob to avoid memcpy

### PR DESCRIPTION
**Pull request by cyyever**

### CODE REVIEW

**Review by datumbox — APPROVED** [review]
The changes seem reasonable to me.

@andfoy have also a look if your don't mind. Thx!

**Comment by andfoy** [comment]
Good to see that `::free` fixes the issues related to ownership when using `from_blob`

**Review by andfoy — APPROVED** [review]

**Comment by fmassa** [comment]
Nice, thanks for doing this and handling the deleter!

File: torchvision/csrc/io/image/cpu/encode_jpeg.cpp [hunk 1]
```diff
@@ -92,16 +92,10 @@ torch::Tensor encode_jpeg(const torch::Tensor& data, int64_t quality) {
...
-  // Copy memory from jpeg buffer, since torch cannot get ownership of it via
-  // `from_blob`
-  auto outPtr = outTensor.data_ptr<uint8_t>();
-  std::memcpy(outPtr, jpegBuf, sizeof(uint8_t) * outTensor.numel());
-
-  free(jpegBuf);
-
-  return outTensor;
+  auto out_tensor =
+      torch::from_blob(jpegBuf, {(long)jpegSize}, ::free, options);
```
Comment (NicolasHug) [inline torchvision/csrc/io/image/cpu/encode_jpeg.cpp:96]: This line is raising the following clang-tidy warning in our internal CI:

> There's an unchecked dereference of an object that is nullable. Make sure you null-check the object before you call methods on it, pass it as non-nullable parameter, or dereference it. If you are sure the the object cannot be nullptr, make it explicit with CHECK_NOTNULL() or assert(). jpegBuf is nullable.

Looks like `jpegBuf` is set above in `jpeg_mem_dest`, but I couldn't find any relevant docs about this function to know what happens if the allocation fails.

Comment (fmassa) [reply torchvision/csrc/io/image/cpu/encode_jpeg.cpp:96]: I think if we have errors the `setjmp` that we do in https://github.com/pytorch/vision/pull/4118/files#diff-dfb505641c632421ae5cc17a4398fe0e3a47c436f62825e88ba2b0aebf497d5eR37 will kick in and the check that `jpegBuf != nullptr`, so maybe this clang-tidy error is a false-positive.

Still let's keep an eye on this

**Pull request merged**